pr_style_review/linter_result.py

```diff
@@ -1,9 +1,66 @@
+import difflib
 from collections import namedtuple
 
 
-LinterResult = namedtuple(
-    'LinterResult', ['level', 'line_number', 'message', 'linter_name',
-                     'file_name'])
+_LinterResult = namedtuple(
+    '_LinterResult', ['level', 'line_number', 'message', 'linter_name',
+                      'file_name'])
+
+
+class LinterResult(_LinterResult):
+    def resolve_commit_id(self, repo, rev='HEAD'):
+        """
+        Resolve commit id from the filename and line_number
+        """
+
+        blame_info_array = repo.blame(rev, self.file_name)
+        line_number = 0
+        for blame_info in blame_info_array:
+            lines = blame_info[1]
+            next_line_number = line_number + len(lines)
+            if self.line_number > line_number and self.line_number <= next_line_number:
+                return blame_info[0].hexsha
+            line_number = next_line_number
+        raise Exception('cannot resolve line')
+
+    def resolve_position(self, repo, commit_id):
+        diffs = repo.commit(repo.commit('{}~'.format(commit_id))).diff(commit_id)
```

Comment:
L26
Either all return statements in a function should return an expression, or none of them should.